kong/tools/request_aware_table.lua

```diff
@@ -0,0 +1,95 @@
+local table_clear = require "table.clear"
+
+local LOG_LEVELS = require "kong.constants".LOG_LEVELS
+local get_log_level = require("resty.kong.log").get_log_level
+
+local get_phase = ngx.get_phase
+local fmt = string.format
+local ngx_var = ngx.var
+
+local NGX_VAR_PHASES = {
+  set = true,
+  rewrite = true,
+  balancer = true,
+  access = true,
+  content = true,
+  header_filter = true,
+  body_filter = true,
+  log = true,
+}
+
+
+local function is_debug_mode()
+  local log_level = get_log_level(LOG_LEVELS[kong.configuration.log_level])
+  return log_level == LOG_LEVELS.debug
```

Comment:
if we decide to ignore the dynamic log level and just use the static one from the config we could avoid repeating this check on every access

Reply:
Let's not do dynamic log level, but put a comment on why this is not needed (CI only feature).